fpga_interchange/arch.h

```diff
@@ -688,6 +702,16 @@ struct Arch : ArchAPI<ArchRanges>
     void place_iobufs(WireId pad_wire, NetInfo *net, const std::unordered_set<CellInfo *> &tightly_attached_bels,
                       std::unordered_set<CellInfo *> *placed_cells);
     void pack_ports();
+
+    // Chain packer
+    void pack_chains();
+    std::unordered_map<IdString, std::vector<IdString>> cluster_driver_cells;
```

Comment:
as a note, now that #718 has just been merged, this will need to be updated to use `dict` instead of `std::unordered_map`

Reply:
Changed all instances into `pool/dict`